### PR TITLE
Eprod 376 set up a minimal canister that registers on the evm testnet

### PR DESCRIPTION
**Pull request by flyq**

### CODE REVIEW

File: src/signature/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+#[tokio::main]
+async fn main() {
+    let wallet = LocalWallet::new(&mut rand::thread_rng());
+    println!("private key: {:?}", wallet.signer().to_bytes().to_vec());
+
+    // ======== generate evm registry tx ========
+    let tx = TransactionRequest::new()
+        .from(wallet.address())
+        .to("0xb0e5863d0ddf7e105e409fee0ecc0123a362e14b") // MINTER_ADDRESS
```
Comment (veeso) [inline src/signature/src/main.rs:11]: this information should be retrieved by calling `registration_ic_agent_info`

File: Cargo.toml [hunk 2]
```diff
@@ -9,26 +9,37 @@ edition = "2021"
...
+dirs = "4.0.0"
+ethereum-types = "0.14"
+ethers = "2.0.6"
 # infinityswap.atlassian.net/browse/EPROD-153 - We are currently depending on a commit, will be removed when new version is released
 ethers-core = { git = "https://github.com/infinity-swap/ethers-rs", default-features = false, package = "ethers-core", branch = "evmc_fork" }
-serde_bytes = "0.11"
-clap = { version = "4.2.4", features = ["derive"] }
-rand = { version = "0.8", features = ["std_rng", "small_rng"] }
+eth-signer = { git = "ssh://git@github.com/infinity-swap/evm-canister", package = "eth-signer" }
+evm_adapter = { git = "ssh://git@github.com/infinity-swap/evm-canister", package = "evm_adapter" }
```
Comment (veeso) [inline Cargo.toml:26]: remove this dependency

**Review by veeso — CHANGES_REQUESTED** [review]

File: src/signature/src/main.rs (hunk 1 above)
Comment (flyq) [reply src/signature/src/main.rs]: fixed

File: Cargo.toml (hunk 2 above)
Comment (flyq) [reply Cargo.toml]: fixed

File: src/signature/src/main.rs [hunk 3]
```diff
@@ -1,14 +1,26 @@
...
+
     let wallet = LocalWallet::new(&mut rand::thread_rng());
     println!("private key: {:?}", wallet.signer().to_bytes().to_vec());
 
     // ======== generate evm registry tx ========
     let tx = TransactionRequest::new()
         .from(wallet.address())
-        .to("0xb0e5863d0ddf7e105e409fee0ecc0123a362e14b") // MINTER_ADDRESS
+        .to(minter_address.as_str()) // MINTER_ADDRESS
         .value(100_000) // REGISTRATION_FEE
```
Comment (veeso) [inline src/signature/src/main.rs:24]: also the registration fee has to be retrieved from the endpoint 😅

**Review by veeso — CHANGES_REQUESTED** [review]

File: src/signature/src/main.rs (hunk 3 above)
Comment (flyq) [reply src/signature/src/main.rs]: fixed

**Review by ufoscout — CHANGES_REQUESTED** [review]
please put all the example code inside an `example` folder

**Review by veeso — APPROVED** [review]

**Review by ufoscout — APPROVED** [review]

**Pull request merged**